Doc/whatsnew/3.12.rst

```diff
@@ -1225,6 +1225,19 @@ Pending Removal in Python 3.14
   :func:`~multiprocessing.set_start_method` APIs to explicitly specify when
   your code *requires* ``'fork'``.  See :ref:`multiprocessing-start-methods`.
 
+* :func:`os.fork` now raises a :exc:`DeprecationWarning` on POSIX when
+  it can detect that it is being called from a multithreaded process.
+  There has always been a fundamental incompatibility with the POSIX
+  platform when doing so.  Even if such code *appeared* to work. We
+  added the warning to to raise awareness as issues encounted by code
+  doing this are becoming more frequent. See the :func:`os.fork`
+  documentation for more details.
+
+  When this warning appears to come from within :mod:`multiprocessing`
+  while using that or :mod:`concurrent.futures` the safe fix is to use a
+  different start method such as ``spawn`` (anywhere) or ``forkserver``
+  (not on macOS).
```

Comment:
Please could we move this to the main section? xref #109766.

A

Reply:
good catch.  I wish I had a sidebar view in my `vim` config for ReST so I could see and outline and which section I was in. :)